src/ast/irbuilderbpf.cpp

```diff
@@ -1669,143 +1644,146 @@ Value *IRBuilderBPF::CreateStrncmp(Value *str1,
+  //   // Explicit check needed for haystack="", needle="" case
+  //   if (needle[0] == '\0')
+  //     return true;
+  //
+  //   for (u64 i = 0; i < haystack_sz && haystack[i] != '\0'; i++) {
+  //     u64 j;
+  //     for (j = 0; j < needle_sz; j++) {
+  //       if (needle[j] == '\0')
+  //         return true;
+  //
+  //       if ((i+j) >= haystack_sz || haystack[i+j] == '\0' || haystack[i+j] != needle[j])
+  //         break;
+  //     }
+  //   }
+  //
+  //   return false;
+  // }
+  //
+  // clang-format on
 
   llvm::Function *parent = GetInsertBlock()->getParent();
-  AllocaInst *store = CreateAllocaBPF(getInt1Ty(), "strcontains.result");
+  BasicBlock *empty_check = BasicBlock::Create(module_.getContext(),
+                                               "strcontains.outer.cond",
+                                               parent);
+  BasicBlock *outer_cond = BasicBlock::Create(module_.getContext(),
+                                              "strcontains.outer.cond",
+                                              parent);
+  BasicBlock *outer_body = BasicBlock::Create(module_.getContext(),
+                                              "strcontains.outer.body",
+                                              parent);
+  BasicBlock *inner_cond = BasicBlock::Create(module_.getContext(),
+                                              "strcontains.inner.cond",
+                                              parent);
+  BasicBlock *inner_body = BasicBlock::Create(module_.getContext(),
+                                              "strcontains.inner.body",
+                                              parent);
+  BasicBlock *inner_notnull = BasicBlock::Create(module_.getContext(),
+                                                 "strcontains.inner.notnull",
+                                                 parent);
+  BasicBlock *inner_incr = BasicBlock::Create(module_.getContext(),
+                                              "strcontains.inner.incr",
+                                              parent);
+  BasicBlock *outer_incr = BasicBlock::Create(module_.getContext(),
+                                              "strcontains.outer.incr",
+                                              parent);
   BasicBlock *done_true = BasicBlock::Create(module_.getContext(),
                                              "strcontains.true",
                                              parent);
   BasicBlock *done_false = BasicBlock::Create(module_.getContext(),
                                               "strcontains.false",
                                               parent);
+  BasicBlock *done = BasicBlock::Create(module_.getContext(),
+                                        "strcontains.done",
+                                        parent);
 
-  CreateStore(getInt1(!inverse), store);
+  AllocaInst *i = CreateAllocaBPF(getInt64Ty(), "strcontain.i");
```

Comment:
To keep consistent naming
```suggestion
  AllocaInst *i = CreateAllocaBPF(getInt64Ty(), "strcontains.i");
```